File: springapp/src/main/java/com/examly/springapp/service/PaymentService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Optional;


import com.examly.springapp.model.Payment;

public interface PaymentService {

     public Payment createPayment(Payment payment);
    public Optional<Payment> getPaymentById(long paymentId);
    public List<Payment> getAllPayments();
    

}

File: angularapp/src/app/services/payment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Global } from '../global';
import { Payment } from '../models/payment.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PaymentService {

  constructor(private http: HttpClient) {}

  public createPayment(payment: Payment): Observable<Payment> {
    return this.http.post(Global.baseUrl + "/api/payments", payment) as Observable<Payment>;
  }

  public getAllPayments(): Observable<Payment[]> {
    return this.http.get(Global.baseUrl + "/api/payments") as Observable<Payment[]>;
  }

  public getPaymentById(paymentId: number): Observable<Payment> {
    return this.http.get(Global.baseUrl + "/api/payments/" + paymentId) as Observable<Payment>;
  }
}

